internal/sqld/sqldwebsockets/websockets.go

```diff
@@ -0,0 +1,230 @@
+		v, err := strconv.ParseInt(val["value"].(string), 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		return v, nil
+	case "text":
+		return val["value"].(string), nil
+	case "blob":
+		base64Encoded := val["value"].(string)
+		v, err := base64.StdEncoding.DecodeString(base64Encoded)
+		if err != nil {
+			return nil, err
+		}
+		return v, nil
+	case "float":
+		return val["value"].(float64), nil
+	}
+	return nil, nil
+}
+
+func (ws *SqldWebsocket) Exec(sql string, params Params, wantRows bool) (*ExecResponse, error) {
+	ctx := context.TODO()
+	requestId := ws.idPool.Get()
+	defer ws.idPool.Put(requestId)
+	stmt := map[string]interface{}{
+		"sql":       sql,
+		"want_rows": wantRows,
+	}
+	if len(params.Values) > 0 {
+		args := []map[string]interface{}{}
+		if len(params.Names) == 0 {
+			for idx := range params.Values {
+				v, err := convertValue(params.Values[idx])
+				if err != nil {
+					return nil, err
+				}
+				args = append(args, v)
+			}
+			stmt["args"] = args
+		} else {
+			for idx := range params.Names {
+				v, err := convertValue(params.Values[idx])
+				if err != nil {
+					return nil, err
+				}
+				arg := map[string]interface{}{
+					"name":  params.Names[idx],
+					"value": v,
+				}
+				args = append(args, arg)
+			}
+			stmt["named_args"] = args
+		}
+	}
+	err := wsjson.Write(ctx, ws.conn, map[string]interface{}{
+		"type":       "request",
+		"request_id": requestId,
+		"request": map[string]interface{}{
+			"type":      "execute",
+			"stream_id": 0,
+			"stmt":      stmt,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var resp interface{}
+	if err = wsjson.Read(ctx, ws.conn, &resp); err != nil {
+		return nil, err
+	}
+
+	if isErrorResp(resp) {
+		err = fmt.Errorf("unable to execute %s: %s", sql, errorMsg(resp))
+		return nil, err
+	}
+
+	return &ExecResponse{resp.(map[string]interface{})["response"].(map[string]interface{})["result"].(map[string]interface{})}, nil
+}
+
+func (ws *SqldWebsocket) Close() error {
+	return ws.conn.Close(websocket.StatusNormalClosure, "All's good")
+}
+
+func Connect(url string, jwt string) (*SqldWebsocket, error) {
+	ctx := context.TODO()
+	c, _, err := websocket.Dial(ctx, url, &websocket.DialOptions{
+		Subprotocols: []string{"hrana1"},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = wsjson.Write(ctx, c, map[string]interface{}{
+		"type": "hello",
+		"jwt":  jwt,
+	})
+	if err != nil {
+		c.Close(websocket.StatusInternalError, err.Error())
+		return nil, err
+	}
+
+	err = wsjson.Write(ctx, c, map[string]interface{}{
+		"type":       "request",
+		"request_id": 0,
+		"request": map[string]interface{}{
+			"type":      "open_stream",
+			"stream_id": 0,
+		},
+	})
+	if err != nil {
+		c.Close(websocket.StatusInternalError, err.Error())
+		return nil, err
+	}
+
+	var helloResp interface{}
+	err = wsjson.Read(ctx, c, &helloResp)
+	if err != nil {
+		c.Close(websocket.StatusInternalError, err.Error())
+		return nil, err
+	}
+	if isErrorResp(helloResp) {
```

Comment:
This is incorrect, the response to a `hello` message has type `hello_ok` or `hello_error`, not `response_error`.

Reply:
Good catch. Fixed. Thanks.